consensus/state.go

```diff
@@ -1202,15 +1203,7 @@ func (cs *State) enterPrecommit(height int64, round int) {
 
 	// +2/3 prevoted nil. Unlock and precommit nil.
 	if len(blockID.Hash) == 0 {
-		if cs.LockedBlock == nil {
-			logger.Info("enterPrecommit: +2/3 prevoted for nil.")
-		} else {
-			logger.Info("enterPrecommit: +2/3 prevoted for nil. Unlocking")
-			cs.LockedRound = -1
-			cs.LockedBlock = nil
-			cs.LockedBlockParts = nil
-			cs.eventBus.PublishEventUnlock(cs.RoundStateEvent())
-		}
```

Comment:
This code is unnecessary as far as I can see because the unlocking happens in `addVote()`

Reply:
we'll need to verify that somehow

Reply:
Yes I would agree - I wouldn't want to have missed something.

Reply:
But this bit of code isn't even covered by the tests and I tried to recreate the conditions but it never happens because the locks change in the `addVote` function beforehand

Reply:
Regardless, there is nothing wrong with keeping the code here. I could just revert it to as it was beforehand and we can review it at a later time